bin/check_mta_sts.py

```diff
@@ -1,77 +1,54 @@
-#   Action output
-
-# Improvements:
-#   Look up domains dynamically
-#   Add this to actions somewhere
-
-# Auth
-AWS_ACCESS_KEY_ID = ""
-AWS_SECRET_ACCESS_KEY = ""
-AWS_SESSION_TOKEN = ""
-
-s3 = boto3.client('s3',
-                  aws_access_key_id=AWS_ACCESS_KEY_ID,
-                  aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
-                  aws_session_token=AWS_SESSION_TOKEN)
-
-BASE_URL = "https://s3.amazonaws.com/880656497252."
+# Keep this updated with all MTA-STS domains
+domains = [
+    "ccrc.gov.uk",
+    "cjit.gov.uk",
+    "cshrcasework.justice.gov.uk",
+    "devl.justice.gov.uk",
+    "g.justice.gov.uk",
+    "govfsl.com",
+    "hmiprisons.gov.uk",
+    "hmiprobation.gov.uk",
+    "ima-citizensrights.org.uk",
+    "imb.org.uk",
+    "judicialappointments.gov.uk",
+    "judicialconduct.gov.uk",
+    "judicialombudsman.gov.uk",
+    "judiciary.uk",
+    "justice.gov.uk",
+    "lawcommission.gov.uk",
+    "newsletter.ima-citizensrights.org.uk",
+    "obr.uk",
+    "ospt.gov.uk",
+    "ppo.gov.uk",
+    "publicguardian.gov.uk",
+    "sentencingcouncil.gov.uk",
+    "sentencingcouncil.org.uk",
+    "ukgovwales.gov.uk",
+    "victimscommissioner.org.uk",
+    "yjb.gov.uk",
+    "yjbservicespp.yjb.gov.uk",
+    "youthjusticepp.yjb.gov.uk"
+]
+
+# Define the suffix variable
 SUFFIX = ".well-known/mta-sts.txt"
 
-# Keep this updated with all MTA-STS domains
-domains = ["ccrc.gov.uk",
-           "cjit.gov.uk",
-           "cshrcasework.justice.gov.uk",
-           "devl.justice.gov.uk",
-           "g.justice.gov.uk",
-           "govfsl.com",
-           "hmiprisons.gov.uk",
-           "hmiprobation.gov.uk",
-           "ima-citizensrights.org.uk",
-           "imb.org.uk",
-           "judicialappointments.gov.uk",
-           "judicialconduct.gov.uk",
-           "judicialombudsman.gov.uk",
-           "judiciary.uk",
-           "justice.gov.uk",
-           "lawcommission.gov.uk",
-           "newsletter.ima-citizensrights.org.uk",
-           "obr.uk",
-           "ospt.gov.uk",
-           "ppo.gov.uk",
-           "publicguardian.gov.uk",
-           "sentencingcouncil.gov.uk",
-           "sentencingcouncil.org.uk",
-           "ukgovwales.gov.uk",
-           "victimscommissioner.org.uk",
-           "yjb.gov.uk",
-           "yjbservicespp.yjb.gov.uk",
-           "youthjusticepp.yjb.gov.uk"]
 
-failed_domains = []
+def main():
+    # Initialize the S3Service Client
+    s3_client = S3Service("880656497252", "ministryofjustice")
 
-# Check MTA STS is configured
-for domain in domains:
-    bucket_name = f"880656497252.{domain}"
+    # List to hold domains that do not enforce MTA-STS
+    failed_domains = []
 
-    try:
-        response = s3.get_object(Bucket=bucket_name, Key=SUFFIX)
-        sts_content = response['Body'].read().decode('utf-8')
-        has_enforce = any(line.startswith("mode: enforce")
-                          for line in sts_content.split('\n'))
+    # Iterate over the list of domains
```

Comment:
You don't need comments here. It's obvious what the code is doing.

Reply:
made changes